core/src/test/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/ProgressImplTest.java

```diff
@@ -0,0 +1,60 @@
+package uk.ac.bbsrc.tgac.miso.core.data.workflow.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.AbstractProgressStep.ProgressStepId;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.Progress;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.ProgressStep;
+
+public class ProgressImplTest {
+  private Progress progress;
+
+  @Before
+  public void setUp() {
+    progress = new ProgressImpl();
+    progress.setSteps(Collections.emptyList());
+  }
+
+  @Test
+  public void testSetUnorderedStepsThroughGetter() {
+    ProgressStep step1 = makePoolProgressStep(1);
+    ProgressStep step2 = makePoolProgressStep(2);
+
+    progress.getSteps().add(step2);
+    progress.getSteps().add(step1);
```

Comment:
I think this behaviour is a little unintuitive (we are setting steps without using setSteps), but it's not as verbose as using setSteps. I've also written a test case to ensure this behaviour.